<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CPPCodeGenerationFormBase</class>
 <widget class="QWidget" name="CPPCodeGenerationFormBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>463</width>
    <height>488</height>
   </rect>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <layout class="QGridLayout" name="ui_gridLayoutMain">
     <item row="0" column="0">
      <widget class="QLabel" name="ui_textLabelTitle">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>40</height>
        </size>
       </property>
       <property name="text">
        <string>&lt;p align=&quot;center&quot;&gt;C++ Code Generation&lt;/p&gt;</string>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QTabWidget" name="ui_tabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>1</number>
       </property>
       <widget class="QWidget" name="ui_tabGeneral">
        <attribute name="title">
         <string>General</string>
        </attribute>
        <widget class="QWidget" name="layoutWidget">
         <property name="geometry">
          <rect>
           <x>4</x>
           <y>4</y>
           <width>441</width>
           <height>222</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Class member prefix</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="ui_classMemberPrefixEdit"/>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="QListWidget" name="ui_generalOptionsListWidget"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="ui_tabMethodBodyGeneration">
        <attribute name="title">
         <string>Method Body Generation</string>
        </attribute>
        <layout class="QGridLayout">
         <item row="2" column="0">
          <spacer>
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Expanding</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>410</width>
             <height>113</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="0">
          <layout class="QGridLayout" name="ui_includeFilesList">
           <item row="1" column="1">
            <widget class="KHistoryComboBox" name="ui_listClassHCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="editable">
              <bool>false</bool>
             </property>
             <property name="duplicatesEnabled">
              <bool>false</bool>
             </property>
             <property name="historyItems">
              <stringlist>
               <string>QPtrList</string>
               <string>vector</string>
               <string>vector</string>
              </stringlist>
             </property>
             <property name="currentItem" stdset="0">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>vector</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="ui_textLabelVariable">
             <property name="text">
              <string>&lt;b&gt;Variable&lt;/b&gt;</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="4">
            <widget class="QPushButton" name="ui_browseStringButton">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32767</height>
              </size>
             </property>
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item row="2" column="7">
            <widget class="QCheckBox" name="ui_globalStringCheckBox">
             <property name="text">
              <string/>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="KHistoryComboBox" name="ui_stringClassHCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="editable">
              <bool>false</bool>
             </property>
             <property name="duplicatesEnabled">
              <bool>false</bool>
             </property>
             <property name="historyItems">
              <stringlist>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
               <string>QString</string>
               <string>string</string>
              </stringlist>
             </property>
             <property name="currentItem" stdset="0">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>string</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="6">
            <spacer>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Minimum</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="6">
            <spacer>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Minimum</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="ui_textLabelStringClass">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>&lt;p align=&quot;center&quot;&gt;String&lt;/p&gt;</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="ui_textLabelListClass">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>&lt;p align=&quot;center&quot;&gt;List&lt;/p&gt;</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="7">
            <widget class="QCheckBox" name="ui_globalListCheckBox">
             <property name="text">
              <string/>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="4">
            <widget class="QPushButton" name="ui_browseListButton">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>32</width>
               <height>32767</height>
              </size>
             </property>
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item row="0" column="7">
            <widget class="QLabel" name="ui_textLabelGlobal">
             <property name="text">
              <string>&lt;i&gt;global?&lt;/i&gt;</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="KHistoryComboBox" name="ui_listIncludeFileHistoryCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="historyItems">
              <stringlist>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
               <string>qptrlist.h</string>
               <string>vector</string>
              </stringlist>
             </property>
             <property name="currentItem" stdset="0">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>vector</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="KHistoryComboBox" name="ui_stringIncludeFileHistoryCombo">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="historyItems">
              <stringlist>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
               <string>qstring.h</string>
               <string>string</string>
              </stringlist>
             </property>
             <property name="currentItem" stdset="0">
              <number>0</number>
             </property>
             <item>
              <property name="text">
               <string>string</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="ui_textLabelClassName">
             <property name="text">
              <string>Class name</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="ui_textLabelIncludeFile">
             <property name="text">
              <string>&lt;i&gt;&lt;p align=&quot;center&quot;&gt;Include file&lt;/p&gt;&lt;/i&gt;</string>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="ui_textLabelHelpMethodBodyGen">
           <property name="text">
            <string>Use following for classes in generated code:</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QGroupBox" name="ui_groupBoxDocumentation">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>80</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>100</height>
        </size>
       </property>
       <property name="title">
        <string>Documentation</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="ui_textLabelStyle">
          <property name="text">
           <string>Style:</string>
          </property>
          <property name="wordWrap">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="KComboBox" name="ui_selectCommentStyle">
          <item>
           <property name="text">
            <string>Slash-Slash (//)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Slash-Star (/**  */)</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KHistoryComboBox</class>
   <extends>KComboBox</extends>
   <header>khistorycombobox.h</header>
  </customwidget>
 </customwidgets>
 <includes>
  <include location="local">klineedit.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>ui_browseListButton</sender>
   <signal>clicked()</signal>
   <receiver>CPPCodeGenerationFormBase</receiver>
   <slot>browseClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ui_browseStringButton</sender>
   <signal>clicked()</signal>
   <receiver>CPPCodeGenerationFormBase</receiver>
   <slot>browseClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
